pkg/controllers/subnetport/subnetport_controller.go

```diff
@@ -356,7 +358,17 @@ func (r *SubnetPortReconciler) GetSubnetPathForSubnetPort(obj *v1alpha1.SubnetPo
 
 func (r *SubnetPortReconciler) AllocateSubnetFromSubnetSet(subnetPort *v1alpha1.SubnetPort, subnetSet *v1alpha1.SubnetSet) (string, error) {
 	log.Info("allocating Subnet for SubnetPort", "subnetPort", subnetPort.Name, "subnetSet", subnetSet.Name)
-	subnetPath, err := common.ServiceMediator.GetAvailableSubnet(subnetSet)
+	nsObj := &v1.Namespace{}
+	if err := r.Client.Get(nil, client.ObjectKey{Name: subnetPort.Namespace}, nsObj); err != nil {
+		err = fmt.Errorf("unable to fetch namespace %s", subnetPort.Namespace)
+		log.Error(err, "")
+		return "", err
+	}
+	var tags []model.Tag
+	for k, v := range nsObj.Labels {
+		tags = append(tags, model.Tag{Scope: servicecommon.String(k), Tag: servicecommon.String(v)})
+	}
+	subnetPath, err := common.ServiceMediator.GetAvailableSubnet(subnetSet, tags)
```

Comment:
I think NS labels should be retrieved from SubnetSet side

Reply:
As discussed.